scripts/runcode.jl

```diff
@@ -77,182 +77,197 @@ julia> chamber(composition, end_time, log_volume_km3, InitialConc_H2O, InitialCo
-        rc = rheol_composition_dict[composition_str]
-        param = Param{Float64}(;
-            composition=composition,
-            rheol=rheol,
-            rho_m0=rc.rho_m0,
-            rho_x0=rc.rho_x0,
-            c_m=rc.c_m,
-            c_x=rc.c_x,
-            L_m=rc.L_m,
-        )
-        r = rheol_dict[rheol]
-        if rheol == "new"
-            param.A, param.B, param.G = r.A, r.B, r.G
-        elseif rheol == "old"
-            param.nn, param.AA, param.G, param.M = r.nn, r.AA, r.G, r.M
-        end
-
-        c = ConstantValues{Float64}()
-        param_IC_Finder = ParamICFinder{Float64}()
-        param_saved_var = ParamSaved{Float64}()
-        sw = SW{Int8}()
-        odesetting = OdeSetting{Float64}()
-        erupt_saved = EruptSaved{Float64}()
-
-        # Initial temperature and viscosity profile around chamber
-        param_saved_var.storeSumk = zeros(param.maxn)
-        param_saved_var.storeSumk_2 = zeros(param.maxn)
-        param_saved_var.storeSumk_old = zeros(param.maxn)
-        param_saved_var.storeSumk_2_old = zeros(param.maxn)
-
-        volume_km3 = 10^log_volume_km3                             # range of volume in km3
-        range_radius = 1000 * (volume_km3 / (4 * pi / 3))^(1 / 3)  # range of radius in m
-        V_0 = 4 * pi / 3 * range_radius^3                          # initial volume of the chamber (m^3)
-
-        # thermal gradient
-        param.Tb = c.T_surface + c.T_gradient * depth        # background temperature crust (K)
-
-        # lithostatic pressure
-        P_0 = param.rho_r * c.grav_acc * depth   # initial chamber pressure (Pa)
-        param.P_lit = P_0
-        param.P_lit_0 = P_0
-        if param.single_eruption
-            P_0 = P_0 + param.DP_crit
-        end
-
-        T_0 = find_liq(composition, InitialConc_H2O, InitialConc_CO2, P_0, param.ini_eps_x)
-
-        T_in = T_0 + 50        # Temperature of inflowing magma (K)
-        param.T_in = T_in
-
-        # set the mass inflow rate
-        param.Mdot_in_pass = build_mdot_in(param.fluxing, rc.rho_m0, log_vfr, P_0, T_in)
-
-        rho_g0 = eos_g_rho_g(P_0, T_0)   # initial gas density
-        eps_x0 = crystal_fraction_eps_x(
-            composition, T_0, P_0, InitialConc_H2O, InitialConc_CO2
-        )
-
-        eps_m0 = 1 - eps_x0
-        rho = rc.rho_m0 * eps_m0 + rc.rho_x0 * eps_x0
-        M_tot = V_0 * rho   # Total mass, initial
-        M_co2_0 = InitialConc_CO2 * V_0 * rho   # Total mass of CO2, initial
-        M_h2o_0 = InitialConc_H2O * V_0 * rho   # Total mass of H2O, initial
-
-        # IC Finder parameters
-        param_IC_Finder.Tol = if composition == Silicic()
-            1e-9
-        else
-            1e-8
-        end
-
-        eps_g0, X_co20, C_co2, phase = IC_Finder(
-            composition, M_h2o_0, M_co2_0, M_tot, P_0, T_0, V_0, rc.rho_m0, param_IC_Finder
+    io = open("$path/$datetime.log", "w+")
+    df = with_logger(SimpleLogger(io)) do
```

Comment:
I replaced `global_logger` with `with_logger` to avoid writing log error when parallelization (run multi chamber function at the same time).